app/src/org/commcare/android/database/connect/models/PushNotificationRecord.kt

```diff
@@ -65,12 +72,51 @@ class PushNotificationRecord : Persisted(), Serializable {
         const val META_ACTION = "action"
         const val META_TITLE = "title"
         const val META_BODY = "body"
-        const val META_DATE_TIME = "date_time"
+        const val META_CREATED_DATE = "created_date"
         const val META_CONFIRMATION_STATUS = "confirmation_status"
         const val META_OPPORTUNITY_ID = "opportunity_id"
         const val META_MESSAGE_ID = "message_id"
         const val META_CHANNEL = "channel"
         const val META_PAYMENT_ID = "payment_id"
         const val META_READ_STATUS = "read_status"
+
+        const val META_TIME_STAMP = "timestamp"
+
+
+        fun fromJsonArray(jsonArray: JSONArray): List<PushNotificationRecord> {
+            val records = mutableListOf<PushNotificationRecord>()
+
+            for (i in 0 until jsonArray.length()) {
+                var obj: JSONObject? = null
+                try {
+                    obj = jsonArray.getJSONObject(i)
+                    val record = PushNotificationRecord().apply {
+                        notificationId = obj.getRequiredString(META_NOTIFICATION_ID, i)
+                        title = obj.getRequiredString(META_TITLE, i)
+                        body = obj.getRequiredString(META_BODY, i)
+                        notificationType = obj.optString(META_NOTIFICATION_TYPE, "")
+                        confirmationStatus = obj.optString(META_CONFIRMATION_STATUS, "")
+                        paymentId = obj.optString(META_PAYMENT_ID, "")
+                        readStatus = obj.optBoolean(META_READ_STATUS, false)
+                        val dateString: String = obj.getString(META_TIME_STAMP)
+                        createdDate = DateUtils.parseDateTime(dateString)
+
+                        val dataObj = obj.optJSONObject("data")
+                        dataObj?.let {
+                            connectMessageId = it.optString(META_MESSAGE_ID, "")
+                            channel = it.optString(META_CHANNEL, "")
+                            action = it.getRequiredString(META_ACTION, i)
+                            opportunityId = it.optString(META_OPPORTUNITY_ID, "")
+                        }
+                    }
+                    records.add(record)
+                } catch (e: JSONException) {
+                    val corruptedJson = obj?.toString() ?: "Unknown JSON"
+                    val errorMessage = "Corrupt pn at index $i: $corruptedJson"
+                    Logger.exception(errorMessage, e)
+                }
```

Comment:
can we have this method throw `JsonException` as it is instead (no catch) and have called decide whethwe they want to swallow vs re-throw. I don't think it's alright for us to swallow this mehtod always for example on notification UI screen where we should be just crashing the app instead.